src/core/screen.c

```diff
@@ -1246,20 +1247,25 @@ get_window_pixbuf (MetaWindow *window,
   *width = gdk_pixbuf_get_width (pixbuf);
   *height = gdk_pixbuf_get_height (pixbuf);
 
-  /* Scale pixbuf to max dimension MAX_PREVIEW_SIZE */
+  /* Scale pixbuf to max dimension based on screen sizscreen size */
```

Comment:
I guess you are ignoring multi-monitor setups... For example if I have 3 1280x1024 monitors side by side I would have 3840px screen width. That would be 384px as preview width, no?

Reply:
@muktupavels - yes, good point. I'm working on a fix for this.

Reply:
okay, I changed it to be based on monitor size, rather than screen size